### DISCUSSION TITLE
How to allow camel-cased columns in Naming Strategy on update method

### DISCUSSION THREAD

**Discussion by D-Marc1** [adonisjs/core#2840] [Help] [answered]
I currently have the default setup of snake-cased database columns, and I convert them to camel-cased on serialization. I am using the following code to achieve this with Naming Strategy:

```ts
BaseModel.namingStrategy.serializedName = (_, key: string) => {
  return key
}
```

This works perfectly even on `where()` or `select()` to pass in the column names camel-cased. However, it requires I pass in snake-cased column on `update()`. How can I resolve this?

**Comment by arthurcvm** [comment]
Hi! Please show your migration and model.

**Reply by D-Marc1** [reply]
Hey @arthurcvm, thanks for your reply.

Here's an example of my `users` table:

**Migration**

```ts
this.schema.createTable(this.tableName, (table) => {
  table.increments('id').primary()
  table.string('first_name', 255).notNullable()
  table.string('last_name', 255).notNullable()
  table.string('email', 255).notNullable().unique()
  table.string('password', 180).notNullable()
  table.timestamp('created_at').defaultTo(this.now()).notNullable()
  table.timestamp('updated_at')
})
```

**Model**

```ts
export default class User extends BaseModel {
  @column({ isPrimary: true })
  public id: number

  @column()
  public firstName: string

  @column()
  public lastName: string

  @column()
  public email: string

  @column({ serializeAs: null })
  public password: string

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoUpdate: true })
  public updatedAt: DateTime

  @beforeSave()
  public static async hashPassword(user: User) {
    if (user.$dirty.password) {
      user.password = await Hash.make(user.password)
    }
  }
}
```

**Reply by D-Marc1** [reply]
Let me know if you need more info from me.

**Reply by arthurcvm** [reply]
Hi @D-Marc1, sorry for delay.
I not identify the problem, you can send the controller methods related in problem?

**Reply by D-Marc1** [reply]
@arthurcvm It's really application wide. Like anytime I would do something like `User.query().update({  })`.

**Reply by arthurcvm** [reply]
Hi @D-Marc1, I did not find solution for this.
@thetutlage @RomainLanz pls help!

**Comment by thetutlage** [comment, marked as answer]
I am a little confused with the original question. But keeping that aside, I think this release https://github.com/adonisjs/lucid/releases/tag/v15.0.3 contains a fix for your issue. Please upgrade and verify :)

**Reply by D-Marc1** [reply]
Can confirm that this works. Thanks!
